actioncable/lib/action_cable/channel/base.rb

```diff
@@ -169,6 +170,18 @@ def perform_action(data)
         end
       end
 
+      # This method is called after subscription has been added to the connection.
+      # Send confirmation here to avoid race conditions when client tries to perform actions
+      # right after receiving confirmation.
```

Comment:
This comment feels like it's describing the change more than the method. In fact, we were already sending the confirmation here -- this is the method that decides (via `subscribed`) whether the subscription is going to be accepted or rejected, after all.